contracts/vault/src/execute.rs

```diff
@@ -162,18 +198,35 @@ fn execute_redistribute_funds(deps: DepsMut, env: Env, info: MessageInfo) -> Res
     for (market, allocation_bps) in config.markets_allocation_bps.iter() {
         let amount = excess.multiply_ratio(*allocation_bps, total_bps);
         if !amount.is_zero() {
-            let deposit_msg = WasmMsg::Execute {
-                contract_addr: market.to_string(),
-                msg: to_json_binary(&MarketExecuteMsg::DepositLiquidity {
-                    stake_to_xlp: false,
-                })?,
-                funds: vec![Coin {
-                    denom: config.usdc_denom.clone(),
-                    amount,
-                }],
-            };
+            if config.usdc_denom.starts_with("osmo1") {
+                messages.push(CosmosMsg::Wasm(WasmMsg::Execute {
+                    contract_addr: config.usdc_denom.clone(),
+                    msg: to_json_binary(&Cw20ExecuteMsg::Transfer {
```

Comment:
Instead of doing a transfer+deposit message, you need to use a CW20 send with the DepositLiquidity message provided within the send.